source/module_hamilt/hamiltlcao.h

```diff
@@ -0,0 +1,106 @@
+#ifndef HAMILTLCAO_H
+#define HAMILTLCAO_H
+
+#include "hamilt.h"
+
+#include "src_lcao/LCAO_gen_fixedH.h"
+#include "src_lcao/gint_gamma.h"
+#include "src_lcao/gint_k.h"
+#include "src_lcao/LCAO_matrix.h"
+
+
+namespace ModuleHamilt
+{
+
+//memory for storing Hamiltonian matrix and overlap matrix for one k point
+template<typename T>
+class LocalMatrix
+{
+    public:
+    std::vector<T> hloc;
+    std::vector<T> sloc;
+    size_t size=1;
+    void resize(size_t size_in, T value){size = size_in; hloc.resize(size, value); sloc.resize(size, value);}
+    T* getH(){return hloc.data();}
+    T* getS(){return sloc.data();}
```

Comment:
Why the `get` method here only returns address info without array length?

Reply:
Because in connector of blas, address and array length are different parameters, so we should offer the function for these purpose. Another reason is H and S array has same length.